### PR TITLE
Add system resolver compatibility workaround

### PR DESCRIPTION
**Pull request by dhaavi**
I recommend reviewing the commits individually in order to follow the development process.

### CODE REVIEW

File: network/api.go [hunk 1]
```diff
@@ -192,7 +208,7 @@ func buildNetworkDebugInfoData(debugConns []*Connection) string {
 
 func (conn *Connection) debugInfoLine() string {
 	var connectionData string
-	if conn.ID != "" {
+	if conn.Type == DNSRequest { // conn.ID !=
```
Comment (ppacher) [inline network/api.go]: I guess the comment at the end of the line can be removed

File: network/database.go [hunk 2]
```diff
@@ -27,37 +26,86 @@ type StorageInterface struct {
 	storage.InjectBase
 }
 
+// Database prefixes:
+// Processes:       network:tree/<PID>
+// DNS Requests:    network:tree/<PID>/dns/<ID>
+// IP Connections:  network:tree/<PID>/ip/<ID>
```
Comment (ppacher) [inline network/database.go:32]: Before releasing we need to make sure to NOT BREAK the UI. AFAIK this should not be a problem but we must try that first!

File: network/database.go [hunk 3]
```diff
@@ -27,37 +26,86 @@ type StorageInterface struct {
...
+		fallthrough
+	case 2:
+		var err error
+		if segments[1] == "" {
+			pid = process.UndefinedProcessID
+		} else {
+			pid, err = strconv.Atoi(segments[1])
+			if err != nil {
+				return 0, "", "", false
+			}
```
Comment (ppacher) [inline network/database.go:77]: shouldn't we log that error? We expect to be able to parse it as a number so this is likely something that points to a bigger issue

File: process/profile.go [hunk 4]
```diff
@@ -54,6 +55,20 @@ func (p *Process) GetProfile(ctx context.Context) (changed bool, err error) {
...
+		case "windows":
+			if (p.Path == `C:\Windows\System32\svchost.exe` || p.Path == `C:\Windows\system32\svchost.exe`) &&
+				(strings.Contains(p.SpecialDetail, "Dnscache") || strings.Contains(p.CmdLine, "-k NetworkService")) {
+				profileID = profile.SystemResolverProfileID
+			}
+		case "linux":
+			switch p.Path {
+			case "/lib/systemd/systemd-resolved",
+				"/usr/lib/systemd/systemd-resolved":
+				profileID = profile.SystemResolverProfileID
```
Comment (ppacher) [inline process/profile.go:69]: For multi-arch systems we should also allow systemd-resolved to be below `/lib64` and `/usr/lib64`.

**Review by ppacher — APPROVED** [review]
LGTM. Just some minor changes requested. Did you notice the CI warnings?

File: network/api.go (hunk 1 above)
Comment (dhaavi) [reply network/api.go]: Already fixed in previous commit, you seem to have review an old state, before some small fixes.

File: network/database.go (hunk 3 above)
Comment (dhaavi) [reply network/database.go:77]: Yup. Checked.

Comment (dhaavi) [reply network/database.go:77]: Ah, wrong comment. This will need some reworking anyway, we'll should have a chat about that.

**Pull request merged**